### ISSUE TITLE
[Bug]: False-positive remote ragdoll when hit by NPC

### ISSUE THREAD

**Issue by abriginets** [Yimura/YimMenu#3107]
### Describe the bug

Remote ragdoll protection occurs upon getting hit by an NPC vehicle and blames some player for it. Players can also trigger this when ramming ped with their vehicles.

### Steps To Reproduce

1. Get hit by an NPC vehicle in public session
2. Get notification about remote ragdol from player

### Expected Behavior

Remote ragdoll protection should only kick-in when other menus trying to ragdoll my ped

### Logs

Not sure if it's going to help but here's my cout.log from another issue. It's contents are from the same game session.

[cout.log](https://github.com/YimMenu/YimMenu/files/15287841/cout.log)

### Screenshots / Video

https://github.com/YimMenu/YimMenu/assets/17240507/7d59f7f4-3d44-4c1a-b011-31663b9e4b1d

### Store

Steam

### Language

English

### YimMenu Language

English (US)

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### Coload

- [X] I've tested this without co-loading anything.

**Comment by snowleopardd** [comment]
yeah this happens a lot, but doesn't bother me since its just a notification

**Comment by gir489returns** [comment]
🤔

If it was blocked, why did you get ragdolled? I think that's the bigger question here.

**Comment by Rxann** [comment]
> 🤔
> 
> If it was blocked, why did you get ragdolled? I think that's the bigger question here.

I agree, It does not make sense 🤣

**Comment by rkwapisz** [comment]
> 🤔
> 
> If it was blocked, why did you get ragdolled? I think that's the bigger question here.

It's 100% reproducible. The message seems to always come from whichever player is closest to you.

There is obviously a local ragdoll handler so your client isn't lagging you waiting for physics to happen, but I don't understand why we'd also be getting a message from a nearby player. Maybe some kind of session ack or other check?

**Comment by gir489returns** [comment]
> > 🤔
> > If it was blocked, why did you get ragdolled? I think that's the bigger question here.
> 
> It's 100% reproducible. The message seems to always come from whichever player is closest to you.
> 
> There is obviously a local ragdoll handler so your client isn't lagging you waiting for physics to happen, but I don't understand why we'd also be getting a message from a nearby player. Maybe some kind of session ack or other check?

I think like the OP mentioned, this is just a case of falsely allerting the user to a big fat nothingburger.

**Comment by abriginets** [comment]
> There is obviously a local ragdoll handler so your client isn't lagging you waiting for physics to happen, but I don't understand why we'd also be getting a message from a nearby player. Maybe some kind of session ack or other check?

This also occur if you explode sticky bombs near some players. It would send you notifications of blamed explosions. Not sure if I should create a separate issue because it feels like both of those have the same root cause.

https://streamable.com/b6hcv1

**Comment by Rxann** [comment]
> > There is obviously a local ragdoll handler so your client isn't lagging you waiting for physics to happen, but I don't understand why we'd also be getting a message from a nearby player. Maybe some kind of session ack or other check?
> 
> This also occur if you explode sticky bombs near some players. It would send you notifications of blamed explosions. Not sure if I should create a separate issue because it feels like both of those have the same root cause.
> 
> https://streamable.com/b6hcv1

The explosions are because, well you are being blamed for the explosion. It's other players telling the game that it was you that blew something up and not them.

**Comment by abriginets** [comment]
I thought this is to detect that feature from 2T1, where you can kill/explode some players but put the blame on someone else. If an exploded player dies, the game will display the player's name you blamed and not you.

**Comment by lonelybud** [comment]
adding this might solve the problem need testing
```C++
case eNetworkEvents::RAGDOLL_REQUEST_EVENT:
{
    ....
		if (!PED::IS_PED_RUNNING_RAGDOLL_TASK(self::ped))
			g.reactions.remote_ragdoll.process(plyr);
    ....
```
